circus-train-s3-s3-copier/src/main/java/com/hotels/bdp/circustrain/s3s3copier/aws/JceksAmazonS3ClientFactory.java

```diff
@@ -136,10 +136,13 @@ private AmazonS3 newInstance(
 
   private HadoopAWSCredentialProviderChain getCredentialsProviderChain(String assumedRole) {
     if (assumedRole != null) {
+      LOG.info("Creating credential chain for assuming role {}", assumedRole);
       return new HadoopAWSCredentialProviderChain(createNewConf(conf, assumedRole));
     } else if (security.getCredentialProvider() != null) {
+      LOG.info("Creating credential chain for assuming role with cred provider {}", security.getCredentialProvider());
       return new HadoopAWSCredentialProviderChain(security.getCredentialProvider());
     }
+    LOG.info("Creating empty credential provider chain");
```

Comment:
Comment is wrong - this isn't really "empty", it's an `EC2ContainerCredentialsProviderWrapper`.